File: 217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
	hm := make(map[int]bool)
	for _, v := range nums {
		if hm[v] {
			return true
		}
		hm[v] = true
	}
	return false
}

// @lc code=end

File: 217.contains-duplicate.java
import java.util.HashSet;

/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hs = new HashSet<Integer>();
        for (int n : nums) {
            if (!hs.add(n)) {
                return true;
            }
        }
        return false;
    }
}
// @lc code=end
